Applications/Games/Age of Empires II HD/Steam/script.js

```diff
@@ -1,5 +1,5 @@
 include("engines.wine.quick_script.steam_script");
-include("utils.functions.filesystem.files");
+const {ls, mkdir, fileExists, cat, cp, getFileSize, fileName, lns, remove, touch, writeToFile, createTempFile, createTempDir, chmod, Checksum} = include("utils.functions.filesystem.files");
```

Comment:
This allows us to only "include" the functions we actually require.
Currently I include all defined functionality for simplicity reasons but we can change that later on.

Reply:
For me, it's not obvious why anybody should know that functions are exported in this order. Also: what happens if you add a function in the middle?

Reply:
The order is not important. You can add the "included" functions in any order you want to

Reply:
Don't you have to change all includes then?

Reply:
What do you mean? What I'm doing here is simple destructuring (see also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). 

So according to my understanding it is ok to use 

```javascript
const {x, y} = include("a");
```

in script "X" and 


```javascript
const {y, z} = include("a");
```

in script "Y"

Reply:
From my understanding, `x` is the first exported function, `y` the second etc. So if you add a new function between `x` and `y` to the module `"a"`, `y` will suddenly be the new function.

Reply:
I think you are confusing this with array destructuring. What I'm doing here is object destructuring. In array destructuring the keys are matched not the indices.

You can test this quite easy yourself in node. Try executing the following code in node:

```javascript
const x = {a: "a", b: "b", c: "c"};
// destructure "x"
const {c, a} = x;
// print c and a
console.log(a);
console.log(c);
```

The result will be:

```
a
c
```

Reply:
Ok, then it's fine.